src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/ReferenceShard.java

```diff
@@ -0,0 +1,92 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import com.google.cloud.dataflow.sdk.coders.*;
+import com.google.cloud.dataflow.sdk.values.KV;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.Serializable;
+
+/**
+ * ReferenceShard is section of the reference genome that's used for sharding work for pairing things with
+ * the reference. The primary use case is pairing reads with the reference.
+ * This class designed to be simple a data-storage class with related static utilities.
+ */
+public class ReferenceShard implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final int shardNumber; // shardNumber is zero-based.
+    private final String contig;
+
+    public static int REFERENCE_SHARD_SIZE = 100000; // This value is likely to be tweaked.
```

Comment:
Ack, I didn't notice this wasn't final until everyone else pointed it out.  Very scary the way it is now.